### PR TITLE
Fix kubelet panic in cgroup manager.

### LINKED ISSUE #43059: [k8s.io] ConfigMap should be consumable via environment variable [Conformance] {E2eNode Suite}
https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/logs/ci-kubernetes-node-kubelet/2498/
Failed: [k8s.io] ConfigMap should be consumable via environment variable [Conformance] {E2eNode Suite}

```
/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/e2e/common/configmap.go:382
Expected error:
    <*errors.errorString | 0xc4202685a0>: {
        s: "failed to get logs from pod-configmaps-59ebd4af-087b-11e7-90c9-42010a800019 for env-test: the server could not find the requested resource (get pods pod-configmaps-59ebd4af-087b-11e7-90c9-42010a800019)",
    }
    failed to get logs from pod-configmaps-59ebd4af-087b-11e7-90c9-42010a800019 for env-test: the server could not find the requested resource (get pods pod-configmaps-59ebd4af-087b-11e7-90c9-42010a800019)
not to have occurred
/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/e2e/framework/util.go:2195
```

Previous issues for this test: #42526

### PR DESCRIPTION
**Pull request by Random-Liu**
Fixes https://github.com/kubernetes/kubernetes/issues/42920.
Fixes https://github.com/kubernetes/kubernetes/issues/42875
Fixes #42927 
Fixes #43059

Check the error in walk function, so that we don't use info when there is an error.

@yujuhong @dchen1107 @derekwaynecarr @vishh /cc @kubernetes/sig-node-bugs

### CODE REVIEW

**Comment by Random-Liu** [comment]
@calebamiles Mark 1.6 because this is a kubelet bug fix.

**Comment by Random-Liu** [comment]
@k8s-bot gce etcd3 e2e test this

**Comment by dchen1107** [comment]
Thanks for fixing this so quickly.

File: pkg/kubelet/cm/cgroup_manager_linux.go [hunk 1]
```diff
@@ -449,12 +449,16 @@ func (m *cgroupManagerImpl) Pids(name CgroupName) []int {
 
 		// WalkFunc which is called for each file and directory in the pod cgroup dir
 		visitor := func(path string, info os.FileInfo, err error) error {
+			if err != nil {
+				glog.V(5).Infof("cgroup manager encountered error scanning cgroup path %q", path)
+				return filepath.SkipDir
```
Comment (vishh) [inline pkg/kubelet/cm/cgroup_manager_linux.go]: I wonder if its safe to skip here...
What prompted you to add this logic?

Comment (Random-Liu) [inline pkg/kubelet/cm/cgroup_manager_linux.go]: Because of this https://github.com/kubernetes/kubernetes/issues/42920

Comment (liggitt) [inline pkg/kubelet/cm/cgroup_manager_linux.go]: @vishh would you say it is equally safe as skipping 10 lines later if there's an error reading cgroup data?

**Comment by Random-Liu** [comment]
@k8s-bot gce etcd3 e2e test this

File: pkg/kubelet/cm/cgroup_manager_linux.go (hunk 1 above)
Comment (Random-Liu) [inline pkg/kubelet/cm/cgroup_manager_linux.go]: The problem is that if there is an error, info may be nil. And this will cause kubelet panic.

Comment (derekwaynecarr) [inline pkg/kubelet/cm/cgroup_manager_linux.go]: this makes sense to me.

**Review by derekwaynecarr — APPROVED** [review]

**Comment by derekwaynecarr** [comment]
this also fixes #42927, thanks!

File: pkg/kubelet/cm/cgroup_manager_linux.go (hunk 1 above)
Comment (vishh) [inline pkg/kubelet/cm/cgroup_manager_linux.go]: I suspect systemd is the one that could cause errors in this logic. We do not expect transient errors with cgroupfs. So maybe in the future, we could have a list of well known controllers this logic expects to always work and fail loudly if one of those controllers throws access errors.

**Comment by dchen1107** [comment]
@derekwaynecarr I believe you were referring to https://github.com/kubernetes/kubernetes/issues/42875 

Fixed the pr description too.

File: pkg/kubelet/cm/cgroup_manager_linux.go (hunk 1 above)
Comment (dchen1107) [inline pkg/kubelet/cm/cgroup_manager_linux.go]: @Random-Liu Shouldn't we also print out err before the return? Same for the following errs?

**Comment by Random-Liu** [comment]
@dchen1107 Updated the PR to print out the error and change log level to `V(4)`, so that we could see the transient error in the test log.

@derekwaynecarr Will this be quite spammy?

**Comment by dchen1107** [comment]
@derekwaynecarr I asked @Random-Liu made above change because the original change did fix the kubelet panic, but I worry about it hides some real issues in the system, and leaves tons of uncleaned pod-cgroup behind.

**Comment by vishh** [comment]
Yeah. It might be useful to generate events on cgroup deletion failures for
now.

**Comment by Random-Liu** [comment]
@k8s-bot cvm gce e2e test this

**Comment by Random-Liu** [comment]
Apply LGTM based on https://github.com/kubernetes/kubernetes/pull/42927#issuecomment-286211993

**Comment by ncdc** [comment]
I'd be willing to bet this fixes most or all of the hits here: https://storage.googleapis.com/k8s-gubernator/triage/index.html?text=the%20server%20could%20not%20find%20the%20requested%20resource%20%5C(get#04092e6e21f91a7731ee

**Pull request merged**